app/Domains/Accounts/Actions/Queries/GetAccounts.php

```diff
@@ -2,17 +2,64 @@
 
 declare(strict_types=1);
 
-namespace App\Domains\Accounts\Actions\Queries;
+namespace App\Domains\Accounts\Actions;
 
 use App\Domains\Accounts\Account;
+use Illuminate\Support\Facades\Response;
+use Lorisleiva\Actions\ActionRequest;
 use Lorisleiva\Actions\Concerns\AsAction;
 
 class GetAccounts
 {
     use AsAction;
 
-    public function handle()
+    public function authorize(ActionRequest $request): Response
     {
-        return Account::paginate(10);
+        $user = $request->user();
+        
+        if ($user->has_general_access) {
+            return Response::allow();
+        }
+
+        return Response::deny('You are unauthorized to perform this action');
+    }
+
+    public function handle(?string $search_term, int $per_page): array
```

Comment:
1. The `$per_page` var will be before the `$search_term` var since we can retrieve accounts even without searching it.

2. Also, place default values like this:
`public function handle(?int $per_page = 10, ?string $search_term = '')`

3. The return type is incorrect.